README.md

```diff
@@ -1,2 +1,207 @@
+|`HEALTHZ_TIMEOUT_SECONDS`|❌      |`20`   | Timeout for end-to-end health checks    |
+|`INSECURE_SKIP_VERIFY`   |❌      |`false`| Skip TLS verification for health checks |
+
+### Example Configuration
+
+```yaml
+env:
+  - name: DOWNSTREAM_SERVICE_URL
+    value: "http://my-service:8080"
+  - name: SMEE_CHANNEL_URL
+    value: "http://smee-server/myawesomechannel"
+  - name: HEALTHZ_TIMEOUT_SECONDS
+    value: "30"
+```
+
+## Kubernetes Deployment
+
+### Complete Example
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-app-with-smee
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-app
+  template:
+    metadata:
+      labels:
+        app: my-app
+    spec:
+      containers:
+        # Main smee client container
+        - name: smee-client
+          image: ghcr.io/chmouel/gosmee:latest
+          args:
+            - client
+            - "https://smee.io/myawesomechannel"
+            - "http://localhost:8080"
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 9100
+            initialDelaySeconds: 10
+            periodSeconds: 10
+
+        # Sidecar container
+        - name: sidecar
+          image: quay.io/konflux-ci/smee-sidecar:latest
+          ports:
+            - containerPort: 8080
+              name: relay
+            - containerPort: 9100
+              name: management
+          env:
+            - name: DOWNSTREAM_SERVICE_URL
+              value: "http://localhost:3000"  # Your app
+            - name: SMEE_CHANNEL_URL
+              value: "https://smee.io/myawesomechannel"
+          livenessProbe:
+            httpGet:
+              path: /livez      # Simple process check
+              port: 9100
+            initialDelaySeconds: 5
+            periodSeconds: 10
+
+        # Your application container
+        - name: my-app
+          image: my-app:latest
+          ports:
+            - containerPort: 3000
+```
+
+### Failure Isolation
+
+Ideally, we could have had the same endpoint on the sidecar used for liveness probe in
+all containers (server, client, sidecar), but in cases the health check fails for
+reasons unrelated to the pod itself (e.g. the upstream Smee server is down), all
+containers on the pod can experience cascading restart loops, that will not allow them
+to be up at the same time to allow the health check to recover.
+
+There are 2 ways to address that:
+1. Client deployments: use the **sidecar's** /healthz endpoint as liveness probe for the
+   client and the **sidecar's** /livez endpoint as liveness probe for the sidecar.
+2. Server deployments: Same as client, but we still need the sidecar's /healthz endpoint
+   configured as liveness probe for both the server and the client.
+
+   In this case, we should configure the probe on the server and the client to be
+   configured so that `failureThreshold` x `initialDelaySeconds` is longer than the max
+   backoff penalty on the cluster.
```

Comment:
```suggestion
   In this case, we should configure the probe on the server and the client so that `failureThreshold` x `initialDelaySeconds` is longer than the max
   backoff penalty on the cluster.
```